Modules/audioop.c

```diff
@@ -1903,31 +1923,60 @@ static PyMethodDef audioop_methods[] = {
 };
 
 static int
-audioop_traverse(PyObject *m, visitproc visit, void *arg) {
-    _audioopstate *state = _audioopstate(m);
-    if (state != NULL)
+audioop_traverse(PyObject *m, visitproc visit, void *arg)
+{
+    audioop_state *state = (audioop_state *)PyModule_GetState(m);
+    if (state) {
         Py_VISIT(state->AudioopError);
+    }
     return 0;
 }
+
 static int
-audioop_clear(PyObject *m) {
-    _audioopstate *state = _audioopstate(m);
-    if (state != NULL)
+audioop_clear(PyObject *m)
+{
+    audioop_state *state = (audioop_state *)PyModule_GetState(m);
+    if (state) {
```

Comment:
Is it really possible that state can be NULL when this function is called?

module_clear() doesn't touch module->md_state value. Only module_dealloc() calls PyMem_FREE(m->md_state). But audioop_clear() must not be called on a deallocated module object.

Reply:
I would prefer to see this question answered before approving the PR.

Reply:
Yes, see [the docs](https://docs.python.org/3/c-api/module.html#c.PyModuleDef.m_clear).

Reply:
Is it worth us to change this logic as victor said?

Reply:
I created https://bugs.python.org/issue39824 to propose to change the Python behavior, so modules would not have to handle md_state=NULL anymore.